Plugins/Format Source Code/plugin.swift

```diff
@@ -0,0 +1,93 @@
+import PackagePlugin
+import Foundation
+
+@main
+struct FormatSourceCode: CommandPlugin {
+  func performCommand(
+    context: PluginContext,
+    arguments: [String]
+  ) async throws {
+    let swiftFormatTool = try context.tool(named: "swift-format")
+    
+    // Extract the arguments that specify what targets to format.
+    var argExtractor = ArgumentExtractor(arguments)
+    let targetNames = argExtractor.extractOption(named: "target")
+    let targetsToFormat = try context.package.targets(named: targetNames)
+    
+    let configuration = argExtractor.extractOption(named: "configuration").first
+    
+    for target in targetsToFormat {
```

Comment:
Instead of spawning `swift-format` once for each target, it would be more efficient to collect the directory paths in an array and then pass them all at once in the args array to just do a single invocation.

Reply:
I fixed It.